### PR TITLE
feat!: Convert to using generic composer interface FFI

### PR DESCRIPTION
**Pull request by vezenovm**

### CODE REVIEW

**Comment by phated** [comment]
Marking as draft while we are pointing to a branch.

**Review by kobyhallx — APPROVED** [review]
💯

File: src/composer.rs [hunk 1]
```diff
@@ -2,26 +2,35 @@ use crate::*;
 
 /// # Safety
 /// pippenger must point to a valid Pippenger object
```
Comment (TomAFrench) [inline src/composer.rs:4]: This doesn't take a pippenger pointer anymore.

**Review by TomAFrench — COMMENTED** [review]
I haven't been following this so will leave it to others to approve. One nit though.

File: src/composer.rs (hunk 1 above)
Comment (phated) [inline src/composer.rs:4]:
```suggestion
```

**Review by kobyhallx — APPROVED** [review]

**Pull request merged**